packages/syntax-core/src/RichSelect/RichSelectList.tsx

```diff
@@ -117,6 +121,7 @@ function RichSelectList(props: RichSelectListProps): ReactElement {
     ...richSelectBoxProps
   } = props;
 
+  const id = useId();
```

Comment:
See https://github.com/Cambly/syntax/blob/09370d918f3fbb34cab9307ab318e68e03e874f2/packages/syntax-core/src/TextField/TextField.tsx#L91-L92 - we want similar logic here